src/app/icd/server/ICDManager.h

```diff
@@ -241,6 +241,12 @@ class ICDManager : public ICDListener, public TestEventTriggerHandler
     void OnNetworkActivity() override;
     void OnKeepActiveRequest(KeepActiveFlags request) override;
     void OnActiveRequestWithdrawal(KeepActiveFlags request) override;
+
+#if CHIP_CONFIG_ENABLE_ICD_DSLS
+    void OnSITModeRequest() override;
```

Comment:
It's a minor, but could we do a simple optimization and use single `OnSITModeRequest` method with boolean argument informing if SIT is requested or withdrawn, instead of two separate methods? Alternatively it could be method like `OnDynamicModeSwitch` with `ICDMode` as an argument.

Reply:
@kkasperczyk-no , I tried to be consistent with what existed already around OnKeepActiveRequest()/OnActiveRequestWithdrawal. Ig you have a strong preference for your proposal, please let me know and I can implement it.

Reply:
No, I'm not going to insist, I just thought it has more sense, but as long as you want to be consistent with the other APIs, it's fine for me.